### PR TITLE
Remap pretrained cls head by class name when fine-tuning

### PR DESCRIPTION
**Pull request by Y-T-G**
Support class weights remapping by matching class names

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Adds class-name-based classification head remapping for detection fine-tuning, improving pretrained weight transfer when datasets have different or reordered class lists 🎯

### 📊 Key Changes
- Added a new `cls_remap` training config option, enabled by default ✅
- During detection model setup, dataset class names are assigned before loading pretrained weights so they can be matched early 🔄
- Introduced logic to remap detection classification head rows by class name when loading weights 🧠
- Supports cases where source and target datasets have overlapping classes but different class counts or class order, such as Objects365 → COCO 📚
- Skips remapping when names are placeholders, no matches exist, or class order is already identical ⚡
- Updates transfer logging to include successfully remapped classification tensors 📊
- Includes a minor test docstring capitalization cleanup 🧹

### 🎯 Purpose & Impact
- Improves fine-tuning from pretrained YOLO detection models across different datasets by preserving useful class-specific knowledge 🚀
- Reduces risk of incorrect class-head weight transfer when class orders differ, which can otherwise hurt training quality ⚠️
- May improve convergence and accuracy for datasets sharing class names with the pretrained model 📈
- Works automatically by default while still allowing users to disable it with `cls_remap=False` if needed 🛠️

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Y-T-G, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This automated message confirms we received your contribution, and an Ultralytics engineer will assist with review soon. Your update for remapping pretrained detection classification heads by class name looks like a useful fine-tuning enhancement ✨

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/nn/tasks.py [hunk 1]
```diff
@@ -332,6 +336,60 @@ def load(self, weights, verbose=True):
...
+        state_dict = self.state_dict()
+        remapped = 0
+        for k in list(csd.keys()):
+            v_src, v_tgt = csd[k], state_dict.get(k)
+            # Only remap cls conv weight/bias: dim 0 is the class count in both, with matching trailing dims
+            if v_tgt is None or v_src.dim() < 1 or v_src.shape[0] != src_nc or v_tgt.shape[0] != tgt_nc:
+                continue
+            if v_src.shape[1:] != v_tgt.shape[1:]:
+                continue
+            v_tgt[valid] = v_src[idx[valid]].to(v_tgt.dtype)
```
Comment (UltralyticsAssistant) [inline ultralytics/nn/tasks.py]: 💡 **MEDIUM**: This remaps every tensor whose first dimension happens to equal `nc`, not just the detection classification convs. That can corrupt unrelated heads when their output count matches the class count (for example auxiliary head tensors in detect-derived models), because those rows are class-order remapped even though their channels are not class logits. Please restrict this loop to the actual cls head parameter keys/modules (e.g. `cv3` / `one2one_cv3` final conv weights and biases) before assigning and popping from `csd`.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The feature is implemented in the right loading path and the config plumbing is straightforward. One correctness issue needs attention: the remap predicate is shape-based and can mutate non-classification tensors that coincidentally have `nc` output channels.

💬 Posted 1 inline comment

File: ultralytics/nn/tasks.py (hunk 1 above)
Comment (Y-T-G) [reply ultralytics/nn/tasks.py]: Fixed in d122c82fa.
